### ISSUE TITLE
Screen reader reads irrelevant info on activating calendar

### ISSUE THREAD

**Issue by v-jogue** [microsoft/fluentui#12867]
Actual result:
On activating select a date combo box, Screen reader announces irrelevant information as "Calendar, calendar dialog  February 2020 Grid‑ Read Only‑ February 24 , 2020 selected‑ Read Only Go to next month March Button‑ To activate press spacebar." Also screen reader announces incorrect role as read only for dates instead of button.

Screen reader should read proper information as "calendar dialog <month>, <year>, <month> <button>, <State>" on all dates in calendar window. Also Role for dates should be announces by JAWS as button instead of ready only. JAWS should announce it as "<focused date, ex-24>, <month>, <year>, button".

Test environment:
Browser: Web Chrome (80.0.3987.116 (Official Build) (64-bit))
OS version: 2004 (OS Build 19564.1005)​
Screen reader: JAWS(2019.1912.1 ILM)

**Comment by joschect** [comment]
@v-jogue Thanks for teh submission! From what I can see this appears to be working correctly with Narrator and Edge. It's possible that the extra information being read out by jaws is a problem with jaws/chrome.

As for the issue with "read-only" I believe that is was an intended behavior since Grid's by default only contained editable fields. That may have changed based on what I'm seeing on the [aria website](https://w3c.github.io/aria-practices/#grid) though.  @lorejoh12 Might know more about that

![image](https://user-images.githubusercontent.com/20801176/80441431-c268ca80-88bf-11ea-8dd2-ae6e24b9c879.png)

**Comment by yangster23** [comment]
Note that I'm facing the same issue with NVDA: 

NVDA is announcing "Go to previous month October  button November 2004  table November 2, 2004  row 2  Tuesday  column 3November 2, 2004  button  row 2"  after expanding "Date of birth" control but here as the NVDA is announcing "Go to previous month October  button" it cause the confusion for screen reader users to think that the screen reader focus is on the "Go to previous month October  button" but in actual the focus is on " November 2, 2004 "
